### PR TITLE
Add guide on Client state and and fix default values of components

### PR DESCRIPTION
**Pull request by abidlabs**
Fixes: https://github.com/gradio-app/gradio/issues/4175

Also sets the default values of several components to `None` instead of an empty string. This makes no difference in the Gradio app itself, but it makes a difference when connecting from the client as any non-None value means that the parameter is not required, which can be confusing if a user did not intentionally specify a non-None value.

### CODE REVIEW

File: guides/08_gradio-clients-and-lite/01_getting-started-with-the-python-client.md [hunk 1]
```diff
@@ -295,3 +295,40 @@ job = client.submit("abcdef")
...
+    textbox = gr.Textbox()
+    number = gr.Number()
+    textbox.submit(count, inputs=[textbox, words], outputs=[number, words])
+    
+demo.launch()
+```
+
+If you were to connect this this Gradio app using the Python Client, you would notice that the API information only shows a single input and output:
+
+```text
```
Comment (aliabd) [inline guides/08_gradio-clients-and-lite/01_getting-started-with-the-python-client.md:322]: if you want a codeblock with no syntax highlighting (like terminal output) use csv

Comment (aliabd) [reply guides/08_gradio-clients-and-lite/01_getting-started-with-the-python-client.md:322]:
```suggestion
```csv
```

**Review by aliabd — APPROVED** [review]
LGTM with one comment

File: guides/08_gradio-clients-and-lite/01_getting-started-with-the-python-client.md (hunk 1 above)
Comment (abidlabs) [reply guides/08_gradio-clients-and-lite/01_getting-started-with-the-python-client.md:322]: Nice

**Pull request merged**